### PR TITLE
React SDK - Expose 'server' submodule for server components (Next 13)

### PR DESCRIPTION
**Pull request by bttf**
## The problem

Next 13 introduces a way to support React **server** and **client** components.

Instead of pages implicitly falling back to static or server-side generation when certain requirements are met, Next 13 requires explicit denotation of which components in a component tree are rendered on the server or client:

- all components are server components
- except for those with the `'use client';` directive at the top of the file

Once a component is marked for the client, the child component tree will be forced to be client-side as well. 

This new paradigm introduces some restrictions, such as the restriction of using strictly client-side React APIs (e.g. Context) in server components. Also, 3rd party libraries that access such APIs **cannot** be used in a server component and require being wrapped in a client component to use.

Currently this means users today who want to use GB in a Next 13 app will be forced to wrap it inside a client component.

## Before
Screenshot 2023-05-31 at 3 57 48 PM

Screenshot 2023-05-31 at 3 57 56 PM

## The solution

This PR introduces a quick stopgap to allow users to leverage GrowthBook in a server component context. It exposes a new `server` submodule that only exposes methods from the JavaScript SDK. Since none of the methods exposed in the JS SDK break server-side restrictions, this is allowable for server components in a Next 13 app.

We also include a `'use client';` directive for our React SDK bundle so that users do not need to 'wrap' our code in a client component in order for Next 13 to accept it.

All changes herein are 100% **backwards compatible** with previous versions of Next and React.

# After
Screenshot 2023-05-31 at 3 58 15 PM

Screenshot 2023-05-31 at 3 58 20 PM

# Documentation updates

Screenshot 2023-06-05 at 4 25 39 PM

### CODE REVIEW

File: packages/front-end/components/OpenVisualEditorLink.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ const OpenVisualEditorLink: FC<{
             return false;
           }
 
-          let isExtensionInstalled = false;
```
Comment (bryce-fitzsimons) [inline packages/front-end/components/OpenVisualEditorLink.tsx]: omit?

Comment (bttf) [reply packages/front-end/components/OpenVisualEditorLink.tsx]: You saw nothing

File: packages/sdk-react/src/server.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  FeatureDefinition,
+  FeatureResult,
+  FeatureResultSource,
+  FeatureRule,
+  JSONValue,
+  SubscriptionFunction,
+  Filter,
+  VariationMeta,
+  VariationRange,
+} from "@growthbook/growthbook";
```
Comment (bryce-fitzsimons) [inline packages/sdk-react/src/server.ts:25]: Is there a reason we can't just do something like
```
export  * as server from "@growthbook/growthbook";
```

Comment (bttf) [reply packages/sdk-react/src/server.ts:25]: I'm not sure why we'd want to do that. Could you explain the potential benefit?

The reason I did it this way was to mimic how we currently do it for the React SDK today, just with React client-side methods omitted.

**Pull request closed without merging**